### DISCUSSION TITLE
Warnings on simple static interface

### DISCUSSION THREAD

**Discussion by M1ke** [vimeo/psalm#7900] [Q&A] [answered]
I have an interface for a static factory method, and am struggling to see how to stop Psalm showing warnings:

https://psalm.dev/r/ed9e2857f7

In this case we have an interface that declares a public static returning `static` and then the implementor. If the implementor returns self we get warnings on both the method type and the return line. However if we declare @return self in a docblock, we get an error.

Is there a way to make this structure work without Psalm complaining?

**Comment by AndrolGenhald** [comment, marked as answer]
This is invalid because you could have `Qux extends Baz` and it would inherit the method returning `self`, so it would not return `static` as required. You can either [mark `Baz` as final](https://psalm.dev/r/e6a9ae301e) to rule out this possibility, or use `@psalm-consistent-constructor` and have the method [construct `static` instead of `self`](https://psalm.dev/r/125e315096).

**Reply by M1ke** [reply]
Thanks, those solutions both make sense.
